src/TimeSteppers/TimeSteppers.jl

```diff
@@ -92,12 +103,16 @@ contribution from non-hydrostatic pressure.
 """
 function calculate_tendencies!(tendencies, velocities, tracers, pressures, diffusivities, model)
 
-    calculate_interior_source_terms!(tendencies, model.architecture, model.grid, model.coriolis, model.buoyancy,
-                                     model.surface_waves, model.closure, velocities, tracers, pressures.pHY′,
-                                     diffusivities, model.forcing, model.parameters, model.clock.time)
+    calculate_interior_source_terms!(
+        tendencies, model.architecture, model.grid, model.coriolis, model.buoyancy,
+        model.surface_waves, model.closure, velocities, tracers, pressures.pHY′,
+        diffusivities, model.forcing, model.parameters, model.clock.time
+    )
 
-    calculate_boundary_source_terms!(tendencies, model.boundary_conditions.solution, model.architecture,
-                                     model.grid, boundary_condition_function_arguments(model)...)
+    calculate_boundary_source_terms!(
+        model.timestepper.Gⁿ, model.architecture, model.velocities,
```

Comment:
isnt `tendencies` the same as `model.timestepper.Gⁿ`? what is the intent of using `model.timestepper.Gⁿ` instead of tendencies? This function mixes them, which is unacceptable. We should either use `tendencies` or `model.timestepper.Gⁿ`, but not both.

Reply:
Yeah this is one of the downsides of not having a fully functional field abstraction...

They are not the same because `tendencies` is a `datatuple` while `model.timestepper.Gⁿ` is a named tuple of fields. `fill_halo_regions` now requires fields so some functions want a `datatuple` while others want the latter.

I agree we should not be mixing the two as it's very confusing.

I think if we are committing to developing a functional field abstraction then we should keep the code as is as parts of it, such as the halo filling, already switched to using fields and has benefited, but put explicit comments anywhere that mixes the two explaining what is going on.

Then once we have a functional field abstraction we can get rid of all the comments and datatuples and we won't be mixing data tuples and field tuples anymore.

Reply:
Okay sorry, I didn't fully grasp the implications of putting boundary conditions into fields. So my comments are a bit silly about time stepping here.

One simplification could be to call `datatuple` locally. A single `datatuple` call was a sensible code simplification previously when boundary conditions and fields were divorced. Now that they are tied together I don't think that's the clearest way to write this code anymore.

We may also want to start appending data tuple variable names with `_data` for clarity too?

Reply:
I like the idea of using `_data` affixes but hopefully we try and flesh out a full field abstraction then this won't be necessary. But if it doesn't work out, then we should use this.